libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,295 @@
+            setTimeout(() => {
+                this.checkedChange.emit(this.value);
+            });
+        });
+    }
+
+    /** @hidden
+     * Adds checkbox or removes checkbox from multiSelectModel
+     */
+    private _updateModel(): void {
+        if (this.tristate) {
+            if (!this.corecheckbox.inputLabel.nativeElement.checked) {
+                this.checkboxCurrentValue = this.corecheckbox.values.falseValue;
+            } else {
+                if (this.corecheckbox.checkboxState === 'indeterminate') {
+                    this.checkboxCurrentValue = this.corecheckbox.values.thirdStateValue;
+                } else {
+                    this.checkboxCurrentValue = this.corecheckbox.values.trueValue;
+                }
+            }
+            this.onChange(this.checkboxCurrentValue);
+            this.indeterminateChange.emit(this.checkboxCurrentValue);
+            this.change.emit(this.checkboxCurrentValue);
+        } else if (this.isBinary) {
+            this.checkedChange.emit(this.checkboxCurrentValue);
+            if (this.checkboxCurrentValue && this.value) {
+                // handles binary checkbox with value
+                this.checkedChange.emit(this.value);
+            }
+            this.onChange(this.checkboxCurrentValue);
+            this.change.emit(this.checkboxCurrentValue);
+        } else {
+            // checkbox has been selected
+            if (this.corecheckbox.inputLabel.nativeElement.checked) {
+                this._addValue();
+            } else {
+                this._removeValue();
+            }
+            this.onChange(this.multiSelectModel);
+
+            // for multiSelect checkbox, all checkbox should have same copy of multiSelectModel.
+            if (this.ngControl) {
+                this.ngControl.control.setValue(this.multiSelectModel);
+            }
+            this.change.emit(this.multiSelectModel);
+        }
+    }
+
+    /** @hidden
+     * triggered when checkbox is unchecked, value removed from model
+     */
+    private _removeValue(): void {
+        if (this.multiSelectModel) {
+            this.multiSelectModel = this.multiSelectModel.filter((val: string) => val !== this.value);
+        }
+    }
+
+    /** @hidden
+     * triggered when checkbox is checked, value added to model
+     */
+    private _addValue(): void {
+        if (this.corecheckbox.checkboxState === 'indeterminate') {
+            this.multiSelectModel = [...this.multiSelectModel, this.checkboxCurrentValue];
+        } else if (this.multiSelectModel) {
+            this.multiSelectModel = [...this.multiSelectModel, this.checkboxCurrentValue];
+        } else {
+            this.multiSelectModel = [this.checkboxCurrentValue];
+        }
+    }
+
+    /**
+     * @hidden
+     * @param value , Array or boolean and string/null for tristate
+     * setting core checkbox control value
+     */
+    private _setCoreCheckboxControl(value: any): void {
```

Comment:
Not sure what is this method doing. The name of method is confusing..

Reply:
solution: 
	1. this function is initialising checkbox with value passed from controller.
	2. renamed to _initialiseCheckboxWithControl